### ISSUE TITLE
您好，怎么在COCO的预训练模型上进行迁移学习啊

### ISSUE THREAD

**Issue by APeiZou** [RangiLyu/nanodet#151]

**Comment by RangiLyu** [comment]
在yml配置文件中取消load_model的注释，并填入coco预训练权重的路径，就可以在coco预训练模型的基础上进行训练

**Comment by Jerryji007** [comment]
请问您有没有试过 finetune， 或者说如果finetune这个模型，应该freeze哪一部分比较好？

**Comment by Jerryji007** [comment]
@RangiLyu

**Comment by APeiZou** [comment]
optimizer:
  lr: 0.14
  momentum: 0.9
  name: SGD
  weight_decay: 0.0001
total_epochs: 3
val_intervals: 10
warmup:
  name: linear
  ratio: 0.1
  steps: 20
[root][02-25 11:47:42]INFO:loaded /data/project_all/project_mmdet/nanodet/nanodet_m.pth, epoch 70
[root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.0.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
[root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.0.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
[root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.1.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
[root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.1.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
[root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.2.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
[root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.2.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
[root][02-25 11:47:42]INFO:Starting training...
[root][02-25 11:47:42]INFO:Start warming up...
@RangiLyu 您好，为什么finetune的时候模型一直这样也不运行也不报错，也没有其它日志打印呢？

**Comment by RangiLyu** [comment]
> optimizer:
> lr: 0.14
> momentum: 0.9
> name: SGD
> weight_decay: 0.0001
> total_epochs: 3
> val_intervals: 10
> warmup:
> name: linear
> ratio: 0.1
> steps: 20
> [root][02-25 11:47:42]INFO:loaded /data/project_all/project_mmdet/nanodet/nanodet_m.pth, epoch 70
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.0.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.0.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.1.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.1.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.2.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.2.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
> [root][02-25 11:47:42]INFO:Starting training...
> [root][02-25 11:47:42]INFO:Start warming up...
> @RangiLyu 您好，为什么finetune的时候模型一直这样也不运行也不报错，也没有其它日志打印呢？

什么系统环境？pytorch版本是多少？ 配置文件里的device设置能否发一下？

**Comment by APeiZou** [comment]
ubuntu18.0.4，pytorch1.7.0，device:
  gpu_ids: [0]
  workers_per_gpu: 8
  batchsize_per_gpu: 32

**Comment by APeiZou** [comment]
ubuntu18.0.4，pytorch1.7.0，device:
gpu_ids: [0]
workers_per_gpu: 8
batchsize_per_gpu: 32
@RangiLyu

**Comment by RangiLyu** [comment]
> ubuntu18.0.4，pytorch1.7.0，device:
> gpu_ids: [0]
> workers_per_gpu: 8
> batchsize_per_gpu: 32
> @RangiLyu 

检查一下启动训练卡住的时候内存有没有被占满？还有就是workers多但是数据放在机械硬盘上也会导致读取数据很慢卡住

**Comment by chaigec123** [comment]
> optimizer:
> lr: 0.14
> momentum: 0.9
> name: SGD
> weight_decay: 0.0001
> total_epochs: 3
> val_intervals: 10
> warmup:
> name: linear
> ratio: 0.1
> steps: 20
> [root][02-25 11:47:42]INFO:loaded /data/project_all/project_mmdet/nanodet/nanodet_m.pth, epoch 70
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.0.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.0.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.1.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.1.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.2.weight, required shapetorch.Size([33, 96, 1, 1]), loaded shapetorch.Size([112, 96, 1, 1]).
> [root][02-25 11:47:42]INFO:Skip loading parameter head.gfl_cls.2.bias, required shapetorch.Size([33]), loaded shapetorch.Size([112]).
> [root][02-25 11:47:42]INFO:Starting training...
> [root][02-25 11:47:42]INFO:Start warming up...
> @RangiLyu 您好，为什么finetune的时候模型一直这样也不运行也不报错，也没有其它日志打印呢？
@RangiLyu 我也出现了相同情况， 环境win10  
pytorch1.7.0
device:
  gpu_ids: [0]
  workers_per_gpu: 1
  batchsize_per_gpu: 80

跑了好几个小时，内存一直用1G多，没有使用Gpu

**Comment by feiqinaqian** [comment]
> 

你好，请问fine-tune需要修改配置文件哪些部分

**Comment by feiqinaqian** [comment]
> ubuntu18.0.4，pytorch1.7.0，device: gpu_ids: [0] workers_per_gpu: 8 batchsize_per_gpu: 32

你好，请问fine-tune需要修改配置文件哪些部分
